.github/workflows/regression-report.yml

```diff
@@ -81,30 +99,177 @@ jobs:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Checkout Checkstyle repository
+        uses: actions/checkout@v4
+        with:
+          repository: checkstyle/checkstyle
+          path: .ci-temp/checkstyle
+
+      - name: Find input file
+        id: find_input_file
+        run: |
+          INPUT_FILE="${{ needs.parse_comment.outputs.input_file }}"
+          CHECKSTYLE_DIR=".ci-temp/checkstyle"
+          FILENAME=$(basename "$INPUT_FILE")
+
+          # Remove .java extension if present
+          FILENAME_NO_EXT="${FILENAME%.java}"
+
+          FOUND_FILE=$(find "$CHECKSTYLE_DIR" -name "${FILENAME_NO_EXT}*.java" -type f)
+          if [ -z "$FOUND_FILE" ]; then
+            echo "Error: Could not find input file"
+            exit 1
+          fi
+          ./.ci/append-to-github-output.sh "input_file" "$FOUND_FILE"
+
+      - name: Download DiffTool JAR
+        uses: robinraju/release-downloader@v1.8
+        with:
+          repository: "relentless-pursuit/test-configs"
+          tag: "extractor-1.0.0"
+          fileName: "extractor-all.jar"
+
+      - name: Set up Java 21
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Generate config
+        id: generate_config
         env:
+          DEFAULT_PROJECTS_LINK: ${{ env.DEFAULT_PROJECTS_LINK }}
+        run: |
+          BASE_URL="https://raw.githubusercontent.com/checkstyle/test-configs/main"
+          FOUND_FILE="${{ steps.find_input_file.outputs.input_file }}"
+          BASE_PATH=".ci-temp/checkstyle"
+          OUTPUT_DIR="generated_configs"
+          mkdir -p "$OUTPUT_DIR"
+          
+          echo "Debug: Original FOUND_FILE = $FOUND_FILE"
+          
+          # Remove .ci-temp/checkstyle/ from the beginning of FOUND_FILE
+          RELATIVE_PATH=${FOUND_FILE#.ci-temp/checkstyle/}
+          
+          echo "Debug: Processed RELATIVE_PATH = $RELATIVE_PATH"
+          
+          # Generate a unique name for the config file
+          CONFIG_FILE_NAME=$(basename "$FOUND_FILE" .java)_config.xml
+          CONFIG_FILE_PATH="$OUTPUT_DIR/$CONFIG_FILE_NAME"
+          
+          echo "Debug: CONFIG_FILE_PATH = $CONFIG_FILE_PATH"
+          
+          # Run the extractor
+          java -jar extractor-all.jar "$BASE_PATH" --input-file "$RELATIVE_PATH" "$CONFIG_FILE_PATH"
+
+          # Create a project properties file
+          PROJECT_FILE_PATH="$OUTPUT_DIR/project.properties"
+          curl --fail-with-body -o "$PROJECT_FILE_PATH" "$DEFAULT_PROJECTS_LINK"
+          
+          # Set the config and project file paths
+          ./.ci/append-to-github-output.sh "config_link" "$CONFIG_FILE_PATH"
+          ./.ci/append-to-github-output.sh "projects_link" "$PROJECT_FILE_PATH"
+          
+          cat "$CONFIG_FILE_PATH"
+          cat "$PROJECT_FILE_PATH"
+
+      - name: Upload Config and Project Files
+        uses: actions/upload-artifact@v3
```

Comment:
Do we need this to store files between jobs ?

Reply:
https://levelup.gitconnected.com/github-actions-how-to-share-data-between-jobs-fc1547defc3e

>Caching is used to re-use data/files between jobs or workflows while Artifacts are used to save files after workflow ended.

Reply:
Config and list of properties will be seen in final html report, so no need to have them as artifacts.

Reply:
> Config and list of properties will be seen in final html report, so no need to have them as artifacts.

riginal case ("Github, generate report for AbstractClassName/Example1"):

In this scenario, the configuration files already exist on a remote server (GitHub).
The parse_comment job extracts the URLs to these existing files.
The make_report job can directly download these files using the provided URLs.
No need to generate or upload artifacts because the files are already accessible via URLs.

Reply:
> Do we need this to store files between jobs ?

Yes, for only locally generated files that we do when generating configs. Answered above.

Reply:
yes, we do need to update other code to work with local files.

Reply:
@relentless-pursuit just store results in cache.

Reply:
not done.

Reply:
Done